src/test/java/com/example/moviemanager/controller/MovieControllerTest.java

```diff
@@ -37,129 +49,215 @@ public class MovieControllerTest {
     @Autowired
     private ObjectMapper objectMapper;
 
+    @ParameterizedTest
+    @CsvFileSource(resources = "/validation_test.csv", numLinesToSkip = 1)
+    void returns_505_response_when_user_input_validation_is_failed(String title, String description, int releaseYear,
+                                                                   double rating, String message) throws Exception {
+        // given
+        String requestBody = objectMapper.writeValueAsString(
+                new CreateMovieRequest(title, description, releaseYear, rating));
+
+        // then
+        mockMvc.perform(post(MOVIES_URL)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(requestBody))
+                .andExpect(status().isInternalServerError())
+                .andExpect(content().string(containsString(message)));
+    }
+
     @Test
     void returns_list_of_all_movies() throws Exception {
         // given
         List<Movie> movies = List.of(
```

Comment:
As we talked in person I would improve it as it is used in single place + do it in all other places.